packages/tasit-action/src/contract/Action.js

```diff
@@ -6,32 +6,67 @@ import ConfigLoader from "../ConfigLoader";
 //  and/or MetaTxAction subclasses
 export class Action extends Subscription {
   #provider;
+  #signer;
+  #rawTx;
   #tx;
   #txConfirmations;
   #timeout;
   #lastConfirmationTime;
 
-  constructor(txPromise, provider) {
+  constructor(rawTx, provider, signer) {
     // Provider implements EventEmitter API and it's enough
     //  to handle with transactions events
     super(provider);
 
     const { events } = ConfigLoader.getConfig();
     const { timeout } = events;
 
-    this.#tx = txPromise.then(
+    this.#rawTx = rawTx;
+    this.#signer = signer;
+    this.#timeout = timeout;
+    this.#provider = provider;
+    this.#txConfirmations = 0;
+  }
+
+  _toRaw = async () => {
+    return await this.#rawTx;
+  };
+
+  #signAndSend = async () => {
+    // TODO: Go deep on gas handling.
+    // Without that, VM returns a revert error instead of out of gas error.
+    // See: https://github.com/tasitlabs/TasitSDK/issues/173
+    //
+    // This command isn't enough
+    // const gasLimit = await this.#provider.estimateGas(this.#rawTx);
+    const gasParams = {
+      gasLimit: 7e6,
+      gasPrice: 1e9,
+    };
+
+    const nonce = await this.#provider.getTransactionCount(
+      this.#signer.address
+    );
+
+    let rawTx = await this.#rawTx;
+
+    rawTx = { ...rawTx, nonce, ...gasParams };
+
+    const signedTx = await this.#signer.sign(rawTx);
+
+    this.#tx = await this.#provider.sendTransaction(signedTx).then(
```

Comment:
Any way we can avoid using `.then` here and use another `await` instead?

Reply:
Sure, I've just realized that `Promise(resolve,request)` is the same as `try { await } catch { }`.
Done.

Reply:
Oh, yep!

Great, thanks. Resolving.